### ISSUE TITLE
[iOS] CodePush not linking correctly with RN 0.56.0

### ISSUE THREAD

**Issue by gbachik** [Microsoft/react-native-code-push#1382]
Currently just trying to get a basic project setup using Microsoft AppCenter and CodePush.

### Issue
1. Basic app setup (react init, upload to github, add in appcenter, everythings good)
2. Install codepush with npm then run react-native link react-native-code-push
3. Try to call codepush.sync() and get undefined

Sorry the image is cluttered but I tried to include everything that could be even remotely useful:
![issue.png](https://s15.postimg.cc/vhx0zlzgr/issue.png)

### Reproducible Demo

* Download https://github.com/gbachik/serverless
* npm install then react-native run-ios

### Environment

* react-native-code-push version: 5.4.1
* react-native version: 0.56.0
* iOS/Android/Windows version: iOS 11.4 xCode 9.4.1
* Does this reproduce on a debug build or release build? debug
* Does this reproduce on a simulator, or only on a physical device? both

I'm probably missing something simple but can't seem to figure it out.
Any help is greatly appreciate.
-Thanks

**Comment by alexandergoncharov** [comment]
Hi @gbachik ,
Thanks for reporting!

Thanks for test project! I just downloaded this project, run `npm i`, `react-native link react-native-code-push`, 'react-native run-ios` and successfully got update by clicking on your test button.

I'm not sure what exactly wrong with your app but seems you setup all this correct and I don't see any issues. If your issue continue appears you can try to build and run app from Xcode just for test.

Please let me know if it was helpful or you have any questions or issue still appears.

Thanks,
Alexander

**Comment by gbachik** [comment]
@alexandergoncharov Thats my fault for including the master branch in the report.
I was actually playing around with this last night to see if I could get codepush working and was able to by just following the manual install instructions (but that defeats the purpose of react-native link right!?)

Here's what I found out:
React-native link will work properly for everything except for adding the libraries to xcode.

When following the manual setup instructions I notice that link had correctly done everything from changing the app delegate to adding the right podfile info etc etc

But I had to manually add the codepush xcode project into mine and add libz etc etc

Not sure why that would happen but that seems to be the same issue no matter how many times I setup a microsoft appcenter app with codepush as of now.

**Comment by alexandergoncharov** [comment]
@gbachik , 
Yeah, You should choose setup approach from here: https://github.com/Microsoft/react-native-code-push/blob/master/docs/setup-ios.md
Recommend approach is [RNPM](https://github.com/Microsoft/react-native-code-push/blob/master/docs/setup-ios.md#plugin-installation-ios---rnpm) (`link` command) because this the fastest approach as it makes all manually steps automatically.
It is strange if `Link` command miss some steps.
Could you please test this command on new clear project?
For this you can do next steps:
1. Create  new React-native app: `react-native init test`, and go in project folder `cd test`.
2. Create app in AppCenter: `appcenter apps create -d test -o iOS -p React-Native`, add deployment: `appcenter codepush deployment add -a <userName>/test Staging` and check deployment key: `appcenter codepush deployment list -a <userName>/test`
3. Install CodePush: `yarn add react-native-code-push --save` and link it: `react-native link react-native-code-push`. For deployments key for ios you should use key from  `appcenter codepush deployment list -a <userName>/test` command. If you don't need Android just tap Enter for ignore it.
4. Setup CodePush configuration. For this you can use `App.js` from example: https://github.com/Microsoft/react-native-code-push/blob/master/Examples/CodePushDemoApp/App.js (Just remove 109 line if you don't have Image )
5. Sign your app with XCode.
6. Run `appcenter codepush release-react -a <userName>/test -d Staging` for releasing update.
7. Change App.js file(I recommend to change some text to see that your release was installed).
8. Run your app and try to update. But please note that in debug mode you changes will not be applied as app will download Jsbundle from dev server. For fixing this you can run app in release mode or just close dev server for testing.  

Please let me know if it was helpful or you have some questions.

**Comment by gbachik** [comment]
When I went through the exact steps you provided It all worked A-OK!
Just to be sure I made two more apps using the app center website from scratch and ran the link command and everything wound up in xcode just fine! I'm not sure what I did to cause it not to link the first two apps a tested but it must have been some user error so I'll close this unless I discover anything else!

-Thanks for your help!
